src/retry-options.ts

```diff
@@ -0,0 +1,50 @@
+import * as core from '@actions/core'
+
+export type RetryOptions = {
+  doNotRetry?: number[]
+  enabled?: boolean
+}
+
+export type RequestOptions = {
+  retries?: number
+}
+
+export function getRetryOptions(
+  retries: number,
+  exemptStatusCodes: number[]
+): [RetryOptions, RequestOptions] {
+  if (retries <= 0) {
+    return [{enabled: false}, {}]
+  }
+
+  const retryOptions: RetryOptions = {
+    enabled: true
+  }
+
+  if (exemptStatusCodes.length > 0) {
+    retryOptions.doNotRetry = exemptStatusCodes
+  }
+
+  const requestOptions: RequestOptions = {
+    retries
+  }
+
+  core.info(
```

Comment:
Should this be debug?
Users might not want every step to output this information